posthog/temporal/data_modeling/run_workflow.py

```diff
@@ -577,6 +582,33 @@ async def hogql_table(query: str, team: Team, logger: FilteringBoundLogger):
             yield table_from_py_list(batch, pa_schema)
 
 
+def _transform_unsupported_decimals(batch: pa.Table, logger: FilteringBoundLogger) -> pa.Table:
+    """
+    Transform high-precision decimal columns to types supported by Delta Lake.
+    Delta Lake supports decimal types up to precision 38, but ClickHouse can return
+    Decimal256 types with 76 digit precision.
+    """
+    schema = batch.schema
+    columns_to_cast = {}
+
+    for field in schema:
+        if pa.types.is_decimal(field.type):
+            decimal_type = field.type
+            if hasattr(decimal_type, "precision") and decimal_type.precision > 38:
```

Comment:
If this doesn't type guard accessing the likes of `precision`, then you can do something like this instead, which will:

```
if isinstance(field.type, pa.Decimal128Type) or isinstance(field.type, pa.Decimal256Type):
```

then you'll be able to do `field.type.precision` without any type errors